server/etcdserver/api/etcdhttp/health.go

```diff
@@ -111,18 +113,57 @@ var (
 		Name:      "health_failures",
 		Help:      "The total number of failed health checks",
 	})
+	rootHealthCheckGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "etcd",
+		Subsystem: "server",
+		Name:      "root_health",
+		Help:      "This metric records the result of the root readyz/livez check.",
+	},
+		[]string{"type"},
+	)
+	rootHealthCheckCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "etcd",
+		Subsystem: "server",
+		Name:      "root_health_total",
+		Help:      "This metric records the total result counts of the root readyz/livez checks.",
+	},
+		[]string{"type", "status"},
+	)
+	healthCheckGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "etcd",
+		Subsystem: "server",
+		Name:      "healthcheck",
+		Help:      "This metric records the result of a single healthcheck.",
+	},
+		[]string{"name", "type"},
```

Comment:
```suggestion
		[]string{"type", "name"},
```

`type` should come first and followed by `name`.